pkgs/node/src/chain.zig

```diff
@@ -345,7 +354,16 @@ pub const BeamChain = struct {
             };
         }
 
-        // 5. fc update head
+        // 5. Save block and state to database
+        var batch = try self.db.initWriteBatch();
```

Comment:
abstract it out to a fn writeToDb, also call this after chain event has been emitted not here, we want to do these things in the end, also the writeToDb can fail silently with just logging an error via logger